openapi.yaml

```diff
@@ -1,279 +1,678 @@
+            type: integer
+        - in: query
+          name: filter[abbreviation]
+          description: 'Search by building abbreviation'
+          required: false
+          schema:
+            type: string
+        - $ref: '#/components/parameters/pageNumber'
+        - $ref: '#/components/parameters/pageSize'
+        - in: query
+          name: include
+          required: false
+          schema:
+            type: array
+            items:
+              type: string
+              enum: [services]
       responses:
         '200':
           description: Successful response
-          schema:
-            $ref: '#/definitions/PetsResult'
+          content:
+            'application/json':
+              schema:
+                $ref: '#/components/schemas/LocationSetResult'
         '400':
-          $ref: '#/responses/400'
+          $ref: '#/components/responses/400'
         '500':
-          $ref: '#/responses/500'
-    post:
-      summary: Add a new pet
-      operationId: postPet
+          $ref: '#/components/responses/500'
+  /locations/{locationId}:
+    get:
       tags:
-        - pets
+        - locations
+      description: 'Finds location by ID'
+      operationId: getLocationByID
       parameters:
-        - in: body
-          name: body
+        - name: locationId
+          in: path
+          description: 'ID of location to be searched by'
           required: true
           schema:
-            $ref: '#/definitions/PetPostBody'
+            type: string
       responses:
-        '201':
-          description: Pet successfully created
+        '200':
+          description: Successful response
+          content:
+            'application/json':
+              schema:
+                $ref: '#/components/schemas/LocationResult'
+        '404':
+          $ref: '#/components/responses/404'
+        '500':
+          $ref: '#/components/responses/500'
+  /locations/{locationId}/services:
+    get:
+      tags:
+        - locations
+        - services
+      description: 'Lists services related to a location'
+      operationId: getServices
+      parameters:
+        - name: locationId
+          in: path
+          description: 'ID of location'
+          required: true
           schema:
-            $ref: '#/definitions/PetResult'
-        '400':
-          $ref: '#/responses/400'
-        '409':
-          description: Conflict
+            type: string
+      responses:
+        '200':
+          description: Successful response
+          content:
+            'application/json':
+              schema:
+                $ref: '#/components/schemas/ServicesResultObjects'
+        '404':
+          $ref: "#/components/responses/404"
+        '500':
+          $ref: "#/components/responses/500"
+  /services:
+    get:
+      tags:
+       - services
+      summary: Get services
+      description: 'Gets all `Service` objects. Can be filtered by parameters.'
+      operationId: getServicesList
```

Comment:
Is it an issue to have two paths with the same operationId? Reading the [openapi docs](https://swagger.io/docs/specification/paths-and-operations/) it sounds like it might be intended to be unique on a per-path basis.

Reply:
@jemisonf right, operation ID should be unique per path. I think this path should use getServices since it's the main services endpoint. You can rename the operation ID to something specific for the relation endpoint.

Reply:
Cool. I'll use `getLocationServices`.